samples/excel/85-preview-apis/range-get-range-edge.yaml

```
@@ -0,0 +1,229 @@
+
+          expensesTable.rows.add(null /*add at the end*/, [
+            ["1/1/2017", "The Phone Company", "Communications", "$120"],
+            ["1/2/2017", "Northwind Electric Cars", "Transportation", "$142"],
+            ["1/5/2017", "Best For You Organics Company", "Groceries", "$27"],
+            ["1/10/2017", "Coho Vineyard", "Restaurant", "$33"],
+            ["1/11/2017", "Bellows College", "Education", "$350"],
+            ["1/15/2017", "Trey Research", "Other", "$135"],
+            ["1/15/2017", "Best For You Organics Company", "Groceries", "$97"]
+          ]);
+
+          sheet.getUsedRange().format.autofitColumns();
+          sheet.getUsedRange().format.autofitRows();
+
+          sheet.activate();
+
+          await context.sync();
+        });
+      }
+
+      /** Default helper for invoking an action and handling errors. */
+      async function tryCatch(callback) {
+        try {
+          await callback();
+        } catch (error) {
+          // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+          console.error(error);
+        }
+      }
+    language: typescript
+template:
+    content: |-
+      <section class="ms-font-m">
+        <p>This sample shows two ways to select the edge of the used range, based on the currently selected range.</p>
+      </section>
+      <section class="setup ms-font-m">
+        <h3>Set up</h3>
+        <button id="setup" class="ms-Button">
+          <span class="ms-Button-label">Add sample data</span>
+        </button>
+      </section>
+
+      <section class="samples ms-font-m">
+        <h3>Try it out</h3>
+        <button id="select-E9" class="ms-Button">
+          <span class="ms-Button-label">Select single cell</span>
+        </button>
+        <button id="select-D8-E9" class="ms-Button">
+          <span class="ms-Button-label">Select multiple cells</span>
+        </button>
+        <br>
+        <p>The first type of range edge selection selects the cell at the furthest edge of the current used range, in the directions up or left. This action matches the result of using the	Control+Arrow key keyboard shortcut while a range is selected.</p>
+        <button id="get-range-edge-left" class="ms-Button">
+          <span class="ms-Button-label">Get left edge of used range</span>
+        </button>
+        <button id="get-range-edge-up" class="ms-Button">
+          <span class="ms-Button-label">Get upward edge of used range</span>
+        </button>
+        <br>
+        <p>The second type of range edge selection selects all the cells from the currently selected range to the furthest edge of the used range, in the directions right or down. This action matches the result of using the Control+Shift+Arrow key keyboard shortcut while a range is selected.</p>
+        <button id="get-extended-range-right" class="ms-Button">
+          <span class="ms-Button-label">Get extended used range to the right</span>
```

The Ctrl-Shift-Arrow key commands seem like they're about adding to the selection, so let's highlight that in the button names.
```suggestion
          <span class="ms-Button-label">Extend used range to the right</span>
```